src/wallet/asyncrpcoperation_shieldcoinbase.cpp

```diff
@@ -0,0 +1,451 @@
+// Copyright (c) 2017 The Zcash developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "asyncrpcqueue.h"
+#include "amount.h"
+#include "core_io.h"
+#include "init.h"
+#include "main.h"
+#include "net.h"
+#include "netbase.h"
+#include "rpcserver.h"
+#include "timedata.h"
+#include "util.h"
+#include "utilmoneystr.h"
+#include "wallet.h"
+#include "walletdb.h"
+#include "script/interpreter.h"
+#include "utiltime.h"
+#include "rpcprotocol.h"
+#include "zcash/IncrementalMerkleTree.hpp"
+#include "sodium.h"
+#include "miner.h"
+
+#include <iostream>
+#include <chrono>
+#include <thread>
+#include <string>
+
+#include "asyncrpcoperation_shieldcoinbase.h"
+
+using namespace libzcash;
+
+static int find_output(UniValue obj, int n) {
+    UniValue outputMapValue = find_value(obj, "outputmap");
+    if (!outputMapValue.isArray()) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "Missing outputmap for JoinSplit operation");
+    }
+
+    UniValue outputMap = outputMapValue.get_array();
+    assert(outputMap.size() == ZC_NUM_JS_OUTPUTS);
+    for (size_t i = 0; i < outputMap.size(); i++) {
+        if (outputMap[i].get_int() == n) {
+            return i;
+        }
+    }
+
+    throw std::logic_error("n is not present in outputmap");
+}
+
+AsyncRPCOperation_shieldcoinbase::AsyncRPCOperation_shieldcoinbase(
+        std::vector<ShieldCoinbaseUTXO> inputs,
+        std::string toAddress,
+        CAmount fee,
+        UniValue contextInfo) :
+        inputs_(inputs), fee_(fee), contextinfo_(contextInfo)
+{
+    if (fee_ < 0 || fee > MAX_MONEY) {
```

Comment:
Why does this check two different parameters (one derived from the other, meaning there could be some invisible-to-here divergence between the two)?

Reply:
Fixed typo, should have been fee.